internal/project/project.go

```diff
@@ -81,981 +49,316 @@ const (
 type Project struct {
-	host *projectHostWithCachedFS
-
-	name string
-	kind Kind
-
-	mu                     sync.Mutex
-	initialLoadPending     bool
-	dirty                  bool
-	version                int
-	deferredClose          bool
-	pendingReload          PendingReload
-	dirtyFilePath          tspath.Path
-	hasAddedorRemovedFiles atomic.Bool
-
-	comparePathsOptions tspath.ComparePathsOptions
-	currentDirectory    string
-	// Inferred projects only
-	rootPath tspath.Path
-
-	configFileName string
-	configFilePath tspath.Path
-	// rootFileNames was a map from Path to { NormalizedPath, ScriptInfo? } in the original code.
-	// But the ProjectService owns script infos, so it's not clear why there was an extra pointer.
-	rootFileNames     *collections.OrderedMap[tspath.Path, string]
-	rootJSFileCount   int
-	compilerOptions   *core.CompilerOptions
-	typeAcquisition   *core.TypeAcquisition
-	parsedCommandLine *tsoptions.ParsedCommandLine
-	programConfig     *tsoptions.ParsedCommandLine
-	program           *compiler.Program
-	checkerPool       *checkerPool
-
-	typingsCacheMu           sync.Mutex
-	unresolvedImportsPerFile map[*ast.SourceFile][]string
-	unresolvedImports        []string
-	typingsInfo              *TypingsInfo
-	typingFiles              []string
-
-	// Watchers
-	failedLookupsWatch      *watchedFiles[map[tspath.Path]string]
-	affectingLocationsWatch *watchedFiles[map[tspath.Path]string]
-	typingsFilesWatch       *watchedFiles[map[tspath.Path]string]
-	typingsDirectoryWatch   *watchedFiles[map[tspath.Path]string]
-	typingsWatchInvoked     atomic.Bool
+	Kind             Kind
+	currentDirectory string
+	configFileName   string
+	configFilePath   tspath.Path
+
+	dirty         bool
+	dirtyFilePath tspath.Path
+
+	host                            *compilerHost
+	CommandLine                     *tsoptions.ParsedCommandLine
+	commandLineWithTypingsFiles     *tsoptions.ParsedCommandLine
+	commandLineWithTypingsFilesOnce sync.Once
+	Program                         *compiler.Program
+	// The kind of update that was performed on the program last time it was updated.
+	ProgramUpdateKind ProgramUpdateKind
+	// The ID of the snapshot that created the program stored in this project.
+	ProgramLastUpdate uint64
+
+	failedLookupsWatch      *WatchedFiles[map[tspath.Path]string]
+	affectingLocationsWatch *WatchedFiles[map[tspath.Path]string]
+	typingsFilesWatch       *WatchedFiles[map[tspath.Path]string]
+	typingsDirectoryWatch   *WatchedFiles[map[tspath.Path]string]
+
+	checkerPool *checkerPool
+
+	// installedTypingsInfo is the value of `project.ComputeTypingsInfo()` that was
+	// used during the most recently completed typings installation.
+	installedTypingsInfo *ata.TypingsInfo
+	// typingsFiles are the root files added by the typings installer.
+	typingsFiles []string
 }
 
 func NewConfiguredProject(
 	configFileName string,
 	configFilePath tspath.Path,
-	host ProjectHost,
+	builder *projectCollectionBuilder,
```

Comment:
`NewConfiguredProject` is exported, but it takes a `*projectCollectionBuilder`, which isn’t exported. That means code outside the package can’t actually call this function. Was the intent for the constructor to be internal (unexported), or for projectCollectionBuilder to be exported?

Reply:
Two reasons really:
 - from tsgolint, we aren't expecting to be able to call these, but it breaks how we generate our shims (we can always work around this so feel free to tell me to do that 🙂 ), as the inconsistency means that we're trying to reference private types.
 - from a code pov, it doesn't really make sense for one to be public, and not, as it means `NewConfiguredProject` can't be called outside the package, despite it being exported?

That said, feel free to ignore these cmts and i can workaround in the tsgolint side.